mainmatter/introduction/graphics_hardware/index.tex

```diff
@@ -142,29 +142,62 @@ \subsubsection{AMD}
-Some SoC based products may have an optinal internal DRAM installed with them, this gives the benefit of having DRAM directly connected to the chip and making it more efficient to store and retrieve data from.
-The base building block of the SoC architecture are Execution Units(EU). These EUs consists of a mix of simultaneous multi-threading and fine grained interleaved multi-threading. \todo{legit have issues with understanding how this works…}
-\todo{add execution unit}
-%HD Graphics 520 – Skylake %(claus’ cpu)
-%HD Graphics 4000 – Ivy Bridge (michaels cpu)
-%HD Graphics 4600 – Haswell &(anders’ cpu)
-%HD Graphics Family (brandborgs cpu)
 
+For this project we have a HD graphics 4600 and HD graphics 520 available for usage. Which means they run on different architectures, gen8 and gen9 respectively.
+
+Before describing the architecture of the inbuilt \gls{GPU} the entire architecture of a Intel \gls{CPU} with a inbuilt \gls{GPU} will be presented.
+
+\fig{figures/SoC_architecture}{System-On-a-Chip architecture of a intel \gls{CPU} with a embedded \gls{GPU}. Figure taken from \cite{computegen9}}{IntelSoC}{1} 
+
+The architecture is partitioned into three parts; \gls{CPU} cores, Processor Graphics (\gls{GPU}), and a ring interconnect domain. 
+The SoC architecture was designed to easily change the components out so it can be used to design for other devices than only desktop computers.
+The ring interconnect is vital for intel multi-core \glspl{CPU} as they communicate to each other through the ring. 
+Embedded \glspl{GPU} are also connected to the rest of the Intel \gls{CPU} through this ring.
+The on-die ring interconnect bus between the \gls{CPU} cores, caches, and embedded \gls{GPU}. 
+Each connected component has its own interface for how it should use the ring (additionally, each \gls{CPU} core is considered as its own unique component). 
+The SoC ring interconnect is a bi-directional ring which has a 32-byte wide data bus.  
+When the \gls{CPU} needs to interact with off-chip systems, this is facilitated by the ring as well. 
+
+The architecture of Intel Processor Graphics was designed to be modular in order to make it more scalable. 
+This means it is possible to design Intel Graphics HD units that will work on other devices than desktop computers. 
+The architecture consists of compute components called execution units, execution units are clustered into groups called subslices, and subslices are clustered into slices. 
+Each component will be further explained.
+
+\fig{figures/Intel_ExecutionUnit}{Architecture of a execution unit, figure taken from \cite{computegen9}}{IntelEU}{1} 
+
+\ref{IntelEU} shows the fundamental building block of Generation 8 and 9 \gls{GPU} architecture, the execution unit. 
+The figure shows that each Execution Unit contains 7 threads, and each thread contains 128 SIMD 32-bit registers.  
```

Comment:
SIMD registers? the registers themselves do not handle instructions, so what is the meaning of this?

Reply:
k